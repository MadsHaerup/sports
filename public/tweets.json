[
  {
    "id": "1",
    "image": "/person1.svg",
    "name": "Noah Jackson",
    "tweet": "I use this app every day and it's a great way to keep up to date with what's going on in football. It's a shame it won't rotate into landscape mode like its main competitor"
  },
  {
    "id": "2",
    "image": "/person2.svg",
    "name": "Darren Wilson",
    "tweet": "Good for scores and commentary but the team lineups have stopped working."
  },
  {
    "id": "3",
    "image": "/person3.svg",
    "name": "Michael Akinuli",
    "tweet": "Good for watching the goals and great for highlights instead of waiting till 10:30 before you can watch highlights on March of the day also no need of sky go"
  },
  {
    "id": "4",
    "image": "/person4.svg",
    "name": "Sarah Udoma",
    "tweet": "This app has videos for other premier league teams who play and it also gives me nofications when my favourite team plays"
  },
  {
    "id": "5",
    "image": "/person2.svg",
    "name": "Zara Kanooh",
    "tweet": "This app is brilliant!"
  }
]