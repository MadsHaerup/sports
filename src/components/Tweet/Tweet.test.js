import axios from 'axios';
import tweets from './tweets.json';

jest.mock('axios');

describe("fetching tweets",()=>{

  it("correctly fetches the user names", ()=>{
    const users =  expect(tweets.map(tweet => tweet.name)).toEqual([
      'Noah Jackson',
      'Darren Wilson',
      'Michael Akinuli',
      'Sarah Udoma',
      'Zara Kanooh'
    ]);

    const resp = {data: users};
    axios.get.mockResolvedValue(resp);

    expect(resp.data).toEqual(users);
  })
// ────────────────────────────────────────────────────────────────────────────────

  it("shows the correct image", ()=>{
    const images = expect(tweets.map(tweet => tweet.image)).toEqual([
      'public/person1.svg',
      'public/person2.svg',
      'public/person3.svg',
      'public/person4.svg',
      'public/person5.svg',
    ]);

    const resp = {data: images};
    axios.get.mockResolvedValue(resp);

    expect(resp.data).toEqual(images);
  })
// ────────────────────────────────────────────────────────────────────────────────

  it("shows the correct tweet", ()=>{
    const tweet = expect(tweets.map(tweet => tweet.tweet)).toEqual([
      "I use this app every day and it's a great way to keep up to date with what's going on in football. It's a shame it won't rotate into landscape mode like its main competitor",
      "Good for scores and commentary but the team lineups have stopped working. ",
      "Good for watching the goals and great for highlights instead of waiting till 10:30 before you can watch highlights on March of the day also no need of sky go",
      "This app has videos for other premier league teams who play and it also gives me nofications when my favourite team plays",
      "This app is brilliant!"
    ])

    const resp = {data: tweet};
    axios.get.mockResolvedValue(resp);

    expect(resp.data).toEqual(tweet);
  })
})

