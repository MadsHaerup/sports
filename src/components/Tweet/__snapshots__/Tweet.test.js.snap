// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tweets correctly matches the name from the json file 1`] = `
Array [
  Object {
    "id": "1",
    "image": "public/person1.svg",
    "name": "Noah Jackson",
    "tweet": "I use this app every day and it's a great way to keep up to date with what's going on in football. It's a shame it won't rotate into landscape mode like its main competitor",
  },
  Object {
    "id": "2",
    "image": "public/person2.svg",
    "name": "Darren Wilson",
    "tweet": "Good for scores and commentary but the team lineups have stopped working. ",
  },
  Object {
    "id": "3",
    "image": "public/person3.svg",
    "name": "Michael Akinuli",
    "tweet": "Good for watching the goals and great for highlights instead of waiting till 10:30 before you can watch highlights on March of the day also no need of sky go",
  },
  Object {
    "id": "4",
    "image": "public/person4.svg",
    "name": "Sarah Udoma",
    "tweet": "This app has videos for other premier league teams who play and it also gives me nofications when my favourite team plays",
  },
  Object {
    "id": "5",
    "image": "public/person5.svg",
    "name": "Zara Kanooh",
    "tweet": "This app is brilliant!",
  },
]
`;

exports[`tweets shows the correct image 1`] = `
Array [
  Object {
    "id": "1",
    "image": "public/person1.svg",
    "name": "Noah Jackson",
    "tweet": "I use this app every day and it's a great way to keep up to date with what's going on in football. It's a shame it won't rotate into landscape mode like its main competitor",
  },
  Object {
    "id": "2",
    "image": "public/person2.svg",
    "name": "Darren Wilson",
    "tweet": "Good for scores and commentary but the team lineups have stopped working. ",
  },
  Object {
    "id": "3",
    "image": "public/person3.svg",
    "name": "Michael Akinuli",
    "tweet": "Good for watching the goals and great for highlights instead of waiting till 10:30 before you can watch highlights on March of the day also no need of sky go",
  },
  Object {
    "id": "4",
    "image": "public/person4.svg",
    "name": "Sarah Udoma",
    "tweet": "This app has videos for other premier league teams who play and it also gives me nofications when my favourite team plays",
  },
  Object {
    "id": "5",
    "image": "public/person5.svg",
    "name": "Zara Kanooh",
    "tweet": "This app is brilliant!",
  },
]
`;

exports[`tweets shows the correct tweet 1`] = `
Array [
  Object {
    "id": "1",
    "image": "public/person1.svg",
    "name": "Noah Jackson",
    "tweet": "I use this app every day and it's a great way to keep up to date with what's going on in football. It's a shame it won't rotate into landscape mode like its main competitor",
  },
  Object {
    "id": "2",
    "image": "public/person2.svg",
    "name": "Darren Wilson",
    "tweet": "Good for scores and commentary but the team lineups have stopped working. ",
  },
  Object {
    "id": "3",
    "image": "public/person3.svg",
    "name": "Michael Akinuli",
    "tweet": "Good for watching the goals and great for highlights instead of waiting till 10:30 before you can watch highlights on March of the day also no need of sky go",
  },
  Object {
    "id": "4",
    "image": "public/person4.svg",
    "name": "Sarah Udoma",
    "tweet": "This app has videos for other premier league teams who play and it also gives me nofications when my favourite team plays",
  },
  Object {
    "id": "5",
    "image": "public/person5.svg",
    "name": "Zara Kanooh",
    "tweet": "This app is brilliant!",
  },
]
`;
